src/type_checker/fn_env.rs

```diff
@@ -80,14 +75,12 @@ impl TypedFnEnv {
 
     /// Exits a scoped block.
     pub fn pop(&mut self) {
-        self.current_scope.checked_sub(1).expect("scope bug");
+        self.current_scope = self.current_scope.checked_sub(1).expect("scope bug");
```

Comment:
woop! did we have a bug here o_O?

Reply:
Yes. It was not  assigning the result  after the subtraction resulting in abnormal values of `current_scope` .